oslabs-beta/DockerLocal
# Fetch projects

## Description:
- The project list is read from a saved file when the user logs in. The file is updated whenever the state of the project list changes.   This saves all of the user's projects, along along with the associated data to persist between sessions, along with minor UI improvements.
## Changes I Made:
- Change file path to read and write file and add error handling to configController.ts
- Add user files to .gitignore
- Home.tsx makes fetch request to middleware to read from file on load, then sets state of the project list. A request is made to the back end to save the project list to write to the file on on any changes.
- When the user adds a project (in ProjectSidebar.tsx), the new project is set as active project
- Main window is resizable 
## How to Test:
- npm install and npm start to launch app
- Click toggle isLoggedIn dev button
- Click Add Project and create a project
- The newly created project and any repos should be saved to a file in the user-projects folder and persist when closing the app or toggling the isLoggedIn button
